src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -44,8 +45,24 @@ public class AddPackageReferenceCommandRunner : IPackageReferenceCommandRunner
             }
             packageReferenceArgs.Logger.LogDebug("Project Dependency Graph Read");
 
-            var projectName = dgSpec.Restore.FirstOrDefault();
-            var originalPackageSpec = dgSpec.GetProjectSpec(projectName);
+            var projectFullPath = Path.GetFullPath(packageReferenceArgs.ProjectPath);
+
+            var matchingPackageSpecs = dgSpec
+                .Projects
+                .Where(p => p.RestoreMetadata.ProjectStyle == ProjectStyle.PackageReference && 
+                PathUtility.GetStringComparerBasedOnOS().Equals(Path.GetFullPath(p.RestoreMetadata.ProjectPath), projectFullPath));
```

Comment:
Add .ToArray() or .ToList() on the end of this to force it to enumerate once.  In the current code it will be enumerated 3 times and run all the linq checks multiple times.